### DISCUSSION TITLE
How do you change Markdown to HTML from rich text of a Strapi blog post in Nuxt 3?

### DISCUSSION THREAD

**Discussion by dev23jjl** [nuxt/nuxt#17950] [Questions]
Hi Nuxt 3 Community,

I'm @dev23jjl, and I am an amateur developer who is working on a personal website/blog/portfolio. I decided to use Strapi for the CMS, and I am using it especially for the blog content. Furthermore, I have been able to fetch data in my Nuxt 3 app, and it has been working well by all means. However, when fetching the body content, I realized that the rich text content is displayed raw in Markdown format. I was wondering if there was anything that I could use to do this. I have heard of markdown-it, but I was unsure on going about implementing it as well as I know that the @nuxtjs/markdown-it module is unsupported in Nuxt 3. If I could, how should I go about using markdown-it. Is there any other way that this is achievable. What can I do to change the raw Markdown post into usable HTML elements?

My blog page (~/pages/blog/index.vue) currently looks like this:

![image](https://user-images.githubusercontent.com/32552207/153312087-8d7e21cf-468a-4288-95fd-dd604a471984.png)

Thanks,
@dev23jjl

**Comment by wobsoriano** [comment]
You can manually install [markdown-it](https://github.com/markdown-it/markdown-it) module and do something like this:

```html
<script setup lang="ts">
import MarkdownIt from "markdown-it";

const content = ref("");

onMounted(() => {
  const md = new MarkdownIt();
  content.value = md.render("# markdown-it rulezz!");
});
</script>

<template>
  <div v-html="content" />
</template>
```

or make a plugin out of it so you can reuse it anywhere

```ts
// plugins/markdown.client.ts
import { defineNuxtPlugin } from "#app";

import MarkdownIt from "markdown-it";

export default defineNuxtPlugin(() => {
  const markdown = new MarkdownIt();
  return {
    provide: {
      markdown,
    },
  };
});
```

```html
<script setup lang="ts">
const { $markdown } = useNuxtApp();
const content = ref("");

onMounted(() => {
  content.value = $markdown.render("# markdown-it rulezz!");
});
</script>

<template>
  <div v-html="content" />
</template>
```

or, you can even use this vue 3 markdown-it wrapper

https://github.com/JanGuillermo/vue3-markdown-it

```ts
// plugins/markdown-it.ts
import { defineNuxtPlugin } from '#app'
import Markdown from 'vue3-markdown-it'

export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.vueApp.use(Markdown)
})
```

```html
<script setup>
const source = ref('# Hello World!')
</script>

<template>
  <div>
    <Markdown :source="source" />
  </div>
</template>
```
